spigot/src/main/kotlin/com/mineinabyss/geary/minecraft/actions/DealDamageAction.kt

```diff
@@ -0,0 +1,19 @@
+package com.mineinabyss.geary.minecraft.actions
+
+import com.mineinabyss.geary.ecs.GearyEntity
+import com.mineinabyss.geary.ecs.actions.GearyAction
+import com.mineinabyss.geary.minecraft.components.toBukkit
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import org.bukkit.entity.LivingEntity
+
+@Serializable
+@SerialName("geary:deal_damage")
+public data class DealDamageAction(
+    val damage: Double
```

Comment:
Maybe add a range parameter here which would randomly choose attacks +- that range?

Reply:
How do you imagine this? Like the damage parameter being a range where you specify a min and max, and if you want static damage you set the min and max to the same number?

In that case, what do you think about writing a custom KSerializer for damage which still accepts doubles but deserializes them as a `[x .. x]` range. That way you can write `damage: 5.0` (which deserializes into a `[5.0 .. 5.0]` range) but also `damage: [5.0 .. 10.0]`. On another note, how would you serialize ranges in configs?

Reply:
Honestly might be simpler to define like a median and deviation. So `middle-range..middle+range`